models/reviews.models.js

```diff
@@ -32,15 +32,60 @@ exports.fetchPatchReview = (review_id, inc_votes) => {
     });
 };
 
-exports.fetchReviews = () => {
-    const queryStr = `
+exports.fetchReviews = (sort_by, order = 'DESC', category) => {
+    const validSortBy = ['title', 'designer', 'owner', 'review_body', 'category', 'created_at', 'votes'];
+    const validOrder = ['ASC', 'DESC'];
+    order = order.toUpperCase();
+    const array = [];
+    let queryStr = `
     SELECT reviews.*, COUNT(comments.comment_id)::int AS comment_count 
     FROM reviews
-    LEFT JOIN comments ON reviews.review_id = comments.review_id
-    GROUP BY reviews.review_id
-    ORDER BY created_at DESC;`
+    LEFT JOIN comments ON reviews.review_id = comments.review_id`
+    const queryStr2 = `SELECT * FROM categories WHERE slug = $1`
 
-    return db.query(queryStr).then((response) => {
+    if (!validOrder.includes(order)) {
+        return Promise.reject( { status: 400, msg: "bad request"});
+    };
+
+    if (category) {
+        queryStr += ` WHERE reviews.category = $1`
+        array.push(category)
+    }
+
+    queryStr += ` GROUP BY reviews.review_id`
+
+    if (sort_by) {
+        if (validSortBy.includes(sort_by)) {
+            queryStr += ` ORDER BY ${sort_by} ${order};`
+        } else {
+            return Promise.reject({ status: 400, msg: "bad request" });
+        }
+    } else {
+        queryStr += ` ORDER BY created_at ${order};`
+    }
+
+    const getReviews = db.query(queryStr, array);
+    // const checkCategoryExists = db.query(queryStr2, array)
+    // if(category) {
+    //     return Promise.all([getReviews, checkCategoryExists]).then((promiseArray) => {
+    //         console.log(promiseArray.rows)
+    //         return (promiseArray.rows[0])
+    //     })
+    // } else {
+    //     return getReviews.then((response) => {
+    //         return response.rows;
+    //     })
+    // }
+
+    return getReviews.then((response) => {
+        if (!response.rows.length) {
+            return db.query(`SELECT * FROM categories WHERE slug = $1`, array).then((response) => {
```

Comment:
This can be handled in a check categories function then use a promise.all, rather than nesting this